source/Meadow-Filter-Parse.js

```diff
@@ -100,6 +131,115 @@ const getFilterComparisonOperator = (pFilterOperator) =>
 	return tmpOperator;
 };
 
+/**
+ *
+ * @param {string} encodedType - "encoded" type from the filter-parse request
+ * @return {string} meadow-friendly type (e.g. SQL type that one could cast to)
+ */
+const getDataType = (encodedType) =>
+{
+	let type = 'CHAR';
+	switch (encodedType)
+	{
+		case 'SINT':
+			type = 'SIGNED';
+			break;
+		case 'UINT':
+			type = 'UNSIGNED';
+			break;
+		case 'DD':
+			// base 10, 5 decimal places (should cover 99% of use cases)
+			type = 'DECIMAL(10,5)';
+			break;
+		case 'DATE':
+			type = 'DATE';
+			break;
+		default:
+			break;
+	}
+	return type;
+};
+
+/**
+ * @spec prepares/formats client inputs so they can be safely added to a query string
+ * @param {Array<string>} values - a list of client inputs to a query
+ * @return {Array<string>} the formatted values
+ */
+const prepareQueryValues = (values) =>
+{
+	return values.map((v) =>
+	{
+		// escape quotes
+		return v.replaceAll("'", "\\'").replaceAll('"', '\\"');
+	});
+};
+
+/**
+ * @spec fieldColumn is the target database (table) column; jsonPath is the target path to use for JSON_EXTRACT
+ * @param {object} pFilterStanza - a filter stanza with the properties defined in the interface
+ * @returns {object} structure { fieldColumn, jsonPath }
+ */
+const parseJSONFieldAndPath = (pFilterStanza) =>
+{
+	const splitIndex = pFilterStanza.Field.indexOf('.');
+	if (splitIndex < 1)
+	{
+		const msg = `Invalid format for Field[${pFilterStanza.Field}]`;
+		throw new Error(msg);
```

Comment:
I think this is the main remaining open question in my mind - does it make sense for this to throw? Will that be handled well in the existing layers?

Reply:
>  does it make sense for this to throw?